src/engine/source/router/src/environmentManager.cpp

```diff
@@ -70,157 +78,44 @@ std::optional<base::Error> EnvironmentManager::delEnvironment(const std::string&
-    {
-        names.push_back(name);
-    }
+    names.reserve(m_environments.size());
+    std::transform(m_environments.begin(),
+                   m_environments.end(),
+                   std::back_inserter(names),
+                   [](const auto& pair) { return pair.first; });
 
     return names;
 }
 
-/********************************************************************
- *                  callback API Request
- ********************************************************************/
-
-api::CommandFn EnvironmentManager::apiCallback()
-{
-    return [this](const json::Json params)
-    {
-        api::WazuhResponse response {};
-        const auto action = params.getString("/action");
-
-        if (!action)
-        {
-            response.message("Missing \"action\" parameter");
-        }
-        else if (action.value() == "set")
-        {
-            response = apiSetEnvironment(params);
-        }
-        else if (action.value() == "get")
-        {
-            response = apiGetEnvironment(params);
-        }
-        else if (action.value() == "delete")
-        {
-            response = apiDelEnvironment(params);
-        }
-        else
-        {
-            response.message(fmt::format("Invalid action '{}'", action.value()));
-        }
-        return response;
-    };
-}
-
-api::WazuhResponse EnvironmentManager::apiSetEnvironment(const json::Json& params)
+std::optional<base::Error>
+EnvironmentManager::forwardEvent(const std::string& name, std::size_t instance, base::Event event)
 {
 
-    api::WazuhResponse response {};
-    auto name = params.getString("/name");
-    if (!name)
-    {
-        response.message(
-            "The \"/name\" parameter is missing, it is required and must be a string");
-        return response;
-    }
-
-    auto err = addEnvironment(name.value());
-
-    if (err)
-    {
-        response.message(err.value().message);
-        return response;
-    }
-
-    // TODO: Delete this when support for multiple environments is added
-    // If build OK, then delete other environments
-    for (auto& otherEnv : getAllEnvironments())
-    {
-        if (otherEnv != name.value())
-        {
-            delEnvironment(otherEnv);
-        }
-    }
-
-    // start environment
-    err = startEnvironment(name.value());
-
-    if (err)
-    {
-        response.message(err.value().message);
-        return response;
-    }
-    response.message("Environment created and started");
-
-    return response;
-}
-
-api::WazuhResponse EnvironmentManager::apiGetEnvironment(const json::Json& params)
-{
-    api::WazuhResponse response {};
-
-    // Array of environments
-    json::Json envs;
-    envs.setArray();
-    for (auto& e : getAllEnvironments())
+    std::shared_lock<std::shared_mutex> lock(m_mutex);
+    auto it = m_environments.find(name);
+    if (it == m_environments.end())
     {
-        envs.appendString(e);
+        return base::Error {fmt::format("Environment '{}' doesn't exist.", name)};
```

Comment:
The "." is supposed to be printed on a higher level

Reply:
done